src/Avalonia.Base/Media/Fonts/FontCollectionBase.cs

```diff
@@ -136,46 +142,408 @@ public virtual bool TryCreateSyntheticGlyphTypeface(
+        /// Attempts to add a glyph typeface from the specified font stream.
+        /// </summary>
+        /// <remarks>The method first attempts to create a glyph typeface from the provided font stream.
+        /// If successful, it adds the created glyph typeface to the collection.</remarks>
+        /// <param name="stream">The font stream containing the font data. The stream must be readable and positioned at the beginning of the
+        /// font data.</param>
+        /// <param name="glyphTypeface">When this method returns, contains the created <see cref="IGlyphTypeface"/> instance if the operation
+        /// succeeds; otherwise, <see langword="null"/>.</param>
+        /// <returns><see langword="true"/> if the glyph typeface was successfully created and added; otherwise, <see
+        /// langword="false"/>.</returns>
+        public bool TryAddGlyphTypeface(Stream stream, [NotNullWhen(true)] out IGlyphTypeface? glyphTypeface)
+        {
+            if (!FontManager.Current.PlatformImpl.TryCreateGlyphTypeface(stream, FontSimulations.None, out glyphTypeface))
+            {
+                return false;
+            }
+
+            return TryAddGlyphTypeface(glyphTypeface);
         }
 
-        IEnumerator IEnumerable.GetEnumerator()
+        /// <summary>
+        /// Attempts to add a font source to the font collection.
+        /// </summary>
+        /// <remarks>This method processes the specified font source and attempts to load all available
+        /// fonts from it.  Fonts are added to the collection based on their family name and typographic family name (if
+        /// available). If the <paramref name="source"/> is <see langword="null"/>, the method returns <see
+        /// langword="false"/>.</remarks>
+        /// <param name="source">The URI of the font source to add. This can be a file path, a resource URI, or another valid font source
+        /// URI.</param>
+        /// <returns><see langword="true"/> if at least one font from the specified source was successfully added to the font
+        /// collection;  otherwise, <see langword="false"/>.</returns>
+        public bool TryAddFontSource(Uri source)
         {
-            return GetEnumerator();
+            if (source is null)
+            {
+                return false;
+            }
+
+            var result = false;
+
+            switch (source.Scheme)
+            {
+                case "avares":
+                case "resm":
+                    {
+                        var assetLoader = AvaloniaLocator.Current.GetRequiredService<IAssetLoader>();
+
+                        var fontAssets = FontFamilyLoader.LoadFontAssets(source);
+
+                        foreach (var fontAsset in fontAssets)
+                        {
+                            var stream = assetLoader.Open(fontAsset);
+
+                            if (!FontManager.Current.PlatformImpl.TryCreateGlyphTypeface(stream, FontSimulations.None, out var glyphTypeface))
```

Comment:
Consider storing `FontManager.Current` in a local variable to avoid the lookup in the global locator inside the loop.